internal/controller/utils.go

```diff
@@ -87,3 +99,43 @@ type WithInformerOptionFunc func(
 
 // Same signature as controllerlib.WithInitialEvent().
 type WithInitialEventOptionFunc func(key controllerlib.Key) controllerlib.Option
+
+// BuildCertPoolAuth returns a PEM-encoded CA bundle from the provided spec. If the provided spec is nil, a
+// nil CA bundle will be returned. If the provided spec contains a CA bundle that is not properly
```

Comment:
Could we have taken in an interface and then collapsed this into one func?

Reply:
Nope, because the structs do not have any functions, they are just data structs.